util/util.go

```diff
@@ -26,3 +28,73 @@ func IsValidUUID(id string) bool {
 func GenerateUUID() string {
 	return uuid.New().String()
 }
+
+// distributeEvents() receives elements from deliverChan, and send them to subscribers
+// Sending a `chan interface{}` to addSubscriber adds a new subscriber.
+// It closes the subscriber channel after sending the element.
+// `go DistributeEvents(deliverChan, addSubscriber)` should be called during API initialization.
+// Any subscriber sent to `addSubscriber` should be buffered chan.
+func DistributeEvents(deliverChan <-chan interface{}, addSubscriber chan chan interface{}) {
+	subscribers := make([]chan interface{}, 0)
+	for {
+		select {
+		case element, ok := <-deliverChan:
+			if !ok {
+				return
+			}
+			for _, subscriber := range subscribers {
+				go func(sub chan interface{}) {
+					sub <- element
+					close(sub)
+				}(subscriber)
+			}
+			subscribers = make([]chan interface{}, 0)
+		case sub, ok := <-addSubscriber:
+			if !ok {
+				return
+			}
+			subscribers = append(subscribers, sub)
+		}
+	}
+}
+
+// LongPolling() subscribes to `addSubscriber`, and do long-polling until anything is delivered.
+// It calls `successHandler` if receives a notification.
+// It calls `timeoutHandler` if there's a timeout.
+// `go DistributeEvents(deliverChan, addSubscriber)` must have been called during API initialization.
+func LongPolling(w http.ResponseWriter, timeout time.Duration, addSubscriber chan chan interface{}, successHandler func(interface{}, http.ResponseWriter), timeoutHandler func(http.ResponseWriter)) {
+	notifyChan := make(chan interface{}, 1)
+	addSubscriber <- notifyChan
+	select {
+	case n := <-notifyChan:
+		successHandler(n, w)
+	case <-time.After(timeout):
+		timeoutHandler(w)
+	}
+}
+
+// Debounce() packs all elements received from channel `inChan` within the specified time window to one slice,
+// and send it to channel `outChan` periodically. If nothing is received in the time window, nothing will be sent to `outChan`.
+func Debounce(inChan chan interface{}, outChan chan []interface{}, window time.Duration) {
```

Comment:
So, once you submit the change for confdeploy plugin, you will make it use this, and remove the debounce in that plugin, correct?

Reply:
Yes, branhc `ISSUE-66918282` in apidGatewayConfDeploy is already using `LongPolling` and `DistributeEvents` in apid-core.
`Debounce ` is not used in our current logic. However it would be useful if we change the design in the future, or if we try to improve the performance.